pkg/controllers/nodeclaim/lifecycle/controller.go

```diff
@@ -74,11 +86,35 @@ func NewController(clk clock.Clock, kubeClient client.Client, cloudProvider clou
 	}
 }
 
+func (c *Controller) Register(_ context.Context, m manager.Manager) error {
```

Comment:
How much of this Register function changed from the current Register function? What was added from the termination controller that wasn't in the lifecycle controller already?

Reply:
The only change was the removal of the event filter for NodeClaims. Previously, we were only watching `Create` events. Since we already had the watch on Nodes, we only need this change to capture the events for both controllers.

Reply:
Hmmm, I wonder why the diff was so large then. Makes sense if we are just adding watches on `Update`

Reply:
Oh, probably just because I moved it above the Reconcile function so it didn't split up reconcile and finalize.